aten/src/ATen/native/Math.h

```diff
@@ -2177,4 +2177,417 @@ static inline C10_HOST_DEVICE T calc_log_ndtr(T x) {
+            -3.31612992738871184744e+5,
+            -1.16237097492762307383e+6,
+            -1.72173700820839662146e+6,
+            -8.53555664245765465627e+5,
+    };
+
+    static const T C[] = {
+            -3.51815701436523470549e+2,
+            -1.70642106651881159223e+4,
+            -2.20528590553854454839e+5,
+            -1.13933444367982507207e+6,
+            -2.53252307177582951285e+6,
+            -2.01889141433532773231e+6,
+    };
+
+    T p, q, u, w, z;
+    int i;
+
+    *sign = 1;
+
+    if (!std::isfinite(x)) return x;
+
+    if (x < T(-34.0)) {
+        q = -x;
+
+        w = reentrant_ln_gamma(q, sign);
+
+        p = std::floor(q);
+
+        if (p == q) return std::numeric_limits<T>::infinity();
+
+        i = p;
+
+        *sign = (i & 1) == 0 ? -1 : 1;
+
+        z = q - p;
+
+        if (z > T(0.5)) z = p + T(1.0) - q;
+
+        z = q * std::sin(c10::pi<T> * z);
+
+        if (z == T(0.0)) return std::numeric_limits<T>::infinity();
+
+        return T(1.1447298858494001741434273513530587116472948129153115715136230714) - std::log(z) - w;
+    }
+
+    if (x < T(13.0)) {
+        z = T(1.0);
+        p = T(0.0);
+        u = x;
+
+        while (u >= T(3.0)) {
+            p = p - T(1.0);
+            u = x + p;
+            z = z * u;
+        }
+
+        while (u < T(2.0)) {
+            if (u == T(0.0)) return std::numeric_limits<T>::infinity();
+
+            z = z / u;
+            p = p + T(1.0);
+            u = x + p;
+        }
+
+        if (z < T(0.0)) {
+            *sign = -1;
+            z = -z;
+        } else {
+            *sign = 1;
+        }
+
+        if (u == T(2.0)) return std::log(z);
+
+        x = x + (p - T(2.0));
+
+        T b = 0.0;
+        T c = 0.0;
+
+        for (int index = 0; index <= 5; index++) b = b * x + B[index];
+        for (int index = 0; index <= 5; index++) c = c * x + C[index];
+
+        return std::log(z) + x * b / c;
+    }
+
+    if (x > T(2.556348e+305)) return *sign * std::numeric_limits<T>::infinity();
+
+    q = (x - T(0.5)) * std::log(x) - x + T(0.9189385332046727417803297364056176398613974736377834128171515404);
+
+    if (x > T(100000000.0)) return q;
+
+    p = T(1.0) / (x * x);
+
+    if (x >= T(1000.0)) return q + (((T(7.9365079365079365079365e-4) * p - T(2.7777777777777777777778e-3)) * p + T(0.0833333333333333333333)) / x);
+
+    T a = 0.0;
+
+    for (int index = 0; index <= 4; index++) a = a * p + A[index];
+
+    return q + (a / x);
+}
+
+template<typename T>
+T stirling_gamma_approximation(T x) {
+    static const T P[] = {
+            +7.87311395793093628397e-4,
+            -2.29549961613378126380e-4,
+            -2.68132617805781232825e-3,
+            +3.47222221605458667310e-3,
+            +8.33333333333482257126e-2,
+    };
+
+    if (x >= T(171.624376956302725)) return std::numeric_limits<double>::infinity();
+
+    T p = 0.0;
+
+    for (int index = 0; index <= 4; index++) p = p * (T(1.0) / x) + P[index];
```

Comment:
Egyptian-style curly braces even when there's only a single line in the scope (the internal linter might actually complain about the lack of them)